pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -1060,7 +1060,32 @@ func isKernelPid(pid int) bool {
 	return err != nil && os.IsNotExist(err)
 }
 
+// GetCapacity returns node capacity data for "cpu", "memory", "ephemeral-storage", and "huge-pages*"
+// At present this method is only invoked when introspecting ephemeral storage
 func (cm *containerManagerImpl) GetCapacity() v1.ResourceList {
+	// We store allocatable ephemeral-storage in the capacity property once we Start() the container manager
+	if _, ok := cm.capacity[v1.ResourceEphemeralStorage]; !ok {
+		// If we haven't yet stored the capacity for ephemeral-storage, we can fetch it directly from cAdvisor,
+		// with some retry tolerance in case it hasn't yet started
+		var ret v1.ResourceList
+		err := wait.PollImmediate(3*time.Second, 5*time.Minute, func() (bool, error) {
```

Comment:
Is this timeout based on test results? I'm curious how these retries work when the `--node-status-update-frequency` is 10s. Will retries here affect the kubelet status update for the node?

Reply:
I think the answer is the retries happen when the node is initialized the 1st time and shouldn't affect the kubelet periodic status update, but just checking 😄

Reply:
Good question. `setNodeStatus(node)` is invoked in two paths:

1. When using `initialNode(ctx)` to get a "local" node object representation (i.e., not a node object representation as the cluster sees it by calling the apiserver GET node|LIST nodes endpoint)
  - `initialNode` is used in two places: (1) it is explicitly used when first registering itself with the apiserver (see `kubelet.registerWithAPIServer()` and (2) it is used as a fallback in various kubelet "get node" paths when the kubeclient is not yet setup, or when there's a transient error calling the apiserver using the kubeclient
2. In the various "update node status" flows, e.g., when a volume attachement/detachment reconciliation fires

A read through the code suggests that the regular node status update reconcilers are used by volume attachment/detachment only. I might be missing something though.

In any event, I think the interval and timeout in the retry loop is totally arbitrary at this point, totally open to tuning that.